infra/lib/layer/index.ts

```diff
@@ -0,0 +1,67 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as archive from "@pulumi/archive";
+import * as aws from "@pulumi/aws";
+import * as command from "@pulumi/command";
+import { stack, defaultTags } from "../tags";
+
+export function createPythonLambdaLayer(config: {
+  name: string;
+  bucketId: pulumi.Input<string>;
+}): aws.lambda.LayerVersion {
+  const cmd =
+    "cd ../../api && poetry export -f requirements.txt -o requirements.txt && poetry run pip install \
+    --platform manylinux2014_x86_64 \
+    --target=../infra/aws/python/lib/python3.12/site-packages/ \
+    --implementation cp \
+    --only-binary=:all: --upgrade \
+    -r requirements.txt";
+
+  const poetryLock = archive.getFile({
+    type: "zip",
+    outputPath: "__pythonDeps.zip",
+    sourceFile: "../../api/poetry.lock",
+  });
+
+  const layerFolder = new command.local.Command(
+    `${config.name}-layer-dependencies`,
+    {
+      create: cmd,
+      update: cmd,
+      archivePaths: ["python/**", "!**__pycache__**"],
+      triggers: [poetryLock.then((pLock) => pLock.outputBase64sha256)],
+    }
+  );
+
+  const layerBucketObjectName = `${config.name}-layer-code.zip`;
+
+  // The layer will contain all the dependencies we have installed
+  const bucketObject = new aws.s3.BucketObject(
+    layerBucketObjectName,
+    {
+      bucket: config.bucketId,
+      // sourceHash: layerFolder.archive.apply(a)  => accessLogsBucket.o),
+      source: layerFolder.archive,
+      sourceHash: poetryLock.then((pLock) => pLock.outputBase64sha256),
+      tags: {
+        ...defaultTags,
+        Name: layerBucketObjectName,
+      },
+    },
+    { dependsOn: [layerFolder] }
+  );
+
+  const layerName = `${config.name}-python-deps`;
+  const lambdaLayer = new aws.lambda.LayerVersion(
```

Comment:
Check if a different version is created every time or this one is updated with the new dependencies

Reply:
This has now been set up to create a new version every time.